runtime/lua/vim/lsp/util.lua

```diff
@@ -530,11 +530,7 @@ function M.apply_text_edits(text_edits, bufnr, offset_encoding)
 
   -- Remove final line if needed
   local fix_eol = has_eol_text_edit
-  fix_eol = fix_eol
-    and (
-      api.nvim_buf_get_option(bufnr, 'eol')
-      or (api.nvim_buf_get_option(bufnr, 'fixeol') and not api.nvim_buf_get_option(bufnr, 'binary'))
-    )
+  fix_eol = fix_eol and (vim.bo[bufnr].eol or (vim.bo[bufnr].fixeol and not vim.bo[bufnr].binary))
```

Comment:
hi, should  the overhead of `vim.bo` be a concern?
since vim.bo is so handy, as an user, it encourages me prefer it over `.nvim_set_option_value()`; but `vim.bo`implies a table plus two function object, `vim.bo[bufnr].eol` implies that x2.

Reply:
> hi, should the overhead of vim.bo be a concern?

No

Reply:
This is just pointless micro-optimizing, unless you can prove it makes an observable difference.